apps/bbsync/save.py

```diff
@@ -0,0 +1,59 @@
+from collectors.bzimport.collectors import BugzillaConnector
+from osidb.models import Flaw
+
+from .query import BugzillaQueryBuilder
+
+
+class BugzillaSaver(BugzillaConnector):
+    """
+    Bugzilla flaw bug save handler
+    flaw validity is assumed and not checked
+    """
+
+    class UnsaveableFlaw(Exception):
+        """
+        error caused by attempt to save a flaw which cannot be saved
+        either by its nature or due to the current saver capabilities
+        """
+
+        pass
+
+    def __init__(self, flaw):
+        """
+        init stuff
+        """
+        self.flaw = flaw
+
+    def save(self):
+        """
+        generic save serving as class entry point
+        which calls create or update handler to continue
+        returns an updated flaw instance (without saving)
+        """
+        return self.create() if self.flaw.bz_id is None else self.update()
+
+    def create(self):
+        """
+        create flaw in Bugilla
+        """
+        bqb = BugzillaQueryBuilder(self.flaw)
+        response = self.bz_conn.createbug(bqb.query)
+        self.flaw.meta_attr["bz_id"] = response.id
```

Comment:
Shouldn't we save the flaw afterwards so that the `bz_id` is not lost?

Reply:
The idea is that the Bugzilla save is just one step of the general `flaw.save` so I wanted to leave the rest to be done there so this really relates to Bugzilla only. Otherwise it would be rather a side effect. One option is even to do the save by syncing the changed data back from Bugzilla which would be done completely separately.